### PR TITLE
Vitotronic: Fix #293 and optimise update things

### LINKED ISSUE #293: Startup problems on RPi 2 (binding.Vitotronic)
At startup on a RPi 2 the thing handler (VitotronicThingHandler) can't find the bridge. 
My feeling is, that the timing of the startup of the core system is wrong:
The thing handler for the 'vitotronic:heating:VITOLIGNO:300P' is start before  the bridge is active.
In the loging the bridge is added  after the handler for the thing (300P) is init.
Log from RPi 2:
````
2015-06-27 14:22:35 [DEBUG] [.io.transport.mqtt.MqttService:118  ] - Starting MQTT Service...
2015-06-27 14:22:36 [DEBUG] [.s.s.mapdb.MapDbStorageService:50   ] - Opened MapDB file at '/home/pi/openhab2/./userdata/mapdb/storage.mapdb'.
2015-06-27 14:22:36 [DEBUG] [.c.c.registry.AbstractRegistry:184  ] - Provider 'org.eclipse.smarthome.core.items.ManagedItemProvider' has been added.
2015-06-27 14:22:37 [DEBUG] [.c.c.registry.AbstractRegistry:184  ] - Provider 'org.eclipse.smarthome.core.thing.ManagedThingProvider' has been added.
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:419  ] - Thing handler factory 'VitotronicHandlerFactory' added
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:283  ] - Thing 'vitotronic:bridge:VITOLIGNO' is tracked by ThingManager.
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:386  ] - Creating handler for thing 'vitotronic:bridge:VITOLIGNO'.
2015-06-27 14:22:37 [DEBUG] [.b.v.h.VitotronicBridgeHandler:226  ] - Initializing Vitotronic bridge handler org.openhab.binding.vitotronic.handler.VitotronicBridgeHandler@161e2d4
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:145  ] - Status of vitotronic:bridge:VITOLIGNO changed to OFFLINE
2015-06-27 14:22:37 [DEBUG] [.e.s.config.core.Configuration:88   ] - Setting value (String) 192.168.2.106 to field 'ipAddress' in configuration class org.openhab.binding.vitotronic.VitotronicBindingConfiguration
2015-06-27 14:22:37 [DEBUG] [.e.s.config.core.Configuration:88   ] - Setting value (Integer) 31113 to field 'port' in configuration class org.openhab.binding.vitotronic.VitotronicBindingConfiguration
2015-06-27 14:22:37 [DEBUG] [.e.s.config.core.Configuration:88   ] - Setting value (Integer) 60 to field 'refreshInterval' in configuration class org.openhab.binding.vitotronic.VitotronicBindingConfiguration
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:283  ] - Thing 'vitotronic:heating:VITOLIGNO:300P' is tracked by ThingManager.
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:386  ] - Creating handler for thing 'vitotronic:heating:VITOLIGNO:300P'.
2015-06-27 14:22:37 [DEBUG] [o.b.v.h.VitotronicThingHandler:81   ] - No available bridge handler found for bridge: vitotronic:bridge:VITOLIGNO .
2015-06-27 14:22:37 [DEBUG] [o.b.v.h.VitotronicThingHandler:45   ] - Thing Handler for 300P started
2015-06-27 14:22:37 [DEBUG] [o.b.v.h.VitotronicThingHandler:91   ] - Can't register vitotronic:heating:VITOLIGNO:300P at bridge bridgeHandler is null.
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:71   ] - Thing handler for thing 'vitotronic:bridge:VITOLIGNO' added.
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:196  ] - Assigning handler for thing 'vitotronic:bridge:VITOLIGNO'.
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:71   ] - Thing handler for thing 'vitotronic:heating:VITOLIGNO:300P' added.
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:145  ] - Status of vitotronic:bridge:VITOLIGNO changed to ONLINE
2015-06-27 14:22:37 [DEBUG] [.c.thing.internal.ThingManager:196  ] - Assigning handler for thing 'vitotronic:heating:VITOLIGNO:300P'.
2015-06-27 14:22:37 [DEBUG] [.c.t.internal.ThingLinkManager:298  ] - Adding linked item 'vitotronic_heating_VITOLIGNO_300P_boiler_temp' to channel 'vitotronic:heating:VITOLIGNO:300P:boiler_temp'.
````

The java version on the RPi is:
````
java version "1.8.0"
Java(TM) SE Runtime Environment (build 1.8.0-b132)
Java HotSpot(TM) Client VM (build 25.0-b70, mixed mode)
````

On my develop PC (Windows) under Eclipse it works fine.  The thingManager waits for the handler factory. 
Log from PC/Eclipse:
````
2015-06-27 13:40:02 [DEBUG] [.c.thing.internal.ThingManager:283  ] - Thing 'vitotronic:bridge:VITOLIGNO' is tracked by ThingManager.
2015-06-27 13:40:02 [DEBUG] [.c.thing.internal.ThingManager:359  ] - Not registering a handler at this point since no handler factory for thing 'vitotronic:bridge:VITOLIGNO' found.
2015-06-27 13:40:02 [DEBUG] [.c.thing.internal.ThingManager:283  ] - Thing 'vitotronic:heating:VITOLIGNO:300P' is tracked by ThingManager.
2015-06-27 13:40:02 [DEBUG] [.c.thing.internal.ThingManager:359  ] - Not registering a handler at this point since no handler factory for thing 'vitotronic:heating:VITOLIGNO:300P' found.
2015-06-27 13:40:02 [DEBUG] [.c.thing.internal.ThingManager:283  ] - Thing 'vitotronic:solar:VITOLIGNO:solar' is tracked by ThingManager.
````
Java version:
````
C:\Users\Stefan\Develop\git\openhab2>java -version
java version "1.8.0_25"
Java(TM) SE Runtime Environment (build 1.8.0_25-b18)
Java HotSpot(TM) 64-Bit Server VM (build 25.25-b02, mixed mode)
````

I have try with a self build version (upstream fetch from 23.06) and with a alpha2.0 binary from github also.

### PR DESCRIPTION
**Pull request by steand**
Signed-off-by: Stefan Andres <develop@myandres.de> (github: @steand)

Fix #293  
Optimise channel update (performance): only linked channels will be updated from heating system (adapter)  
Minor changes in things-types.xml

### CODE REVIEW

File: addons/binding/org.openhab.binding.vitotronic/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
         <channels>
             <channel id="systemtime" typeId="systemtime"/>
-            <channel id="outside_temp" typeId="outside_temp"/>
+            <channel id="outsite_temp" typeId="outsite_temp"/>
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.vitotronic/ESH-INF/thing/thing-types.xml]: why did you rename this? "outside" should be the correct spelling.

File: addons/binding/org.openhab.binding.vitotronic/src/main/java/org/openhab/binding/vitotronic/handler/VitotronicBridgeHandler.java [hunk 2]
```diff
@@ -122,7 +122,7 @@ public void registerVitotronicThingListener(VitotronicThingHandler thingHandler)
 				logger.trace("register thingHandler for thing: {}", thingID);
 				updateThingHandlerStatus(thingHandler, this.getStatus());
 				sendSocketData("get " + thingID);
-			}
+			} else logger.trace("thingHandler for thing: '{}' allready registerd", thingID);
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.vitotronic/src/main/java/org/openhab/binding/vitotronic/handler/VitotronicBridgeHandler.java]: add some curly braces for the else branch as well.

File: addons/binding/org.openhab.binding.vitotronic/src/main/java/org/openhab/binding/vitotronic/handler/VitotronicThingHandler.java [hunk 3]
```diff
@@ -63,22 +63,55 @@ public void updateStatus(ThingStatus status ) {
...
+	private synchronized  VitotronicBridgeHandler getBridgeHandler(Bridge bridge) {
+		
+		VitotronicBridgeHandler bridgeHandler = null;
+
 		  ThingHandler handler = bridge.getHandler();
 		  if (handler instanceof VitotronicBridgeHandler) {
 		      bridgeHandler = (VitotronicBridgeHandler) handler;
 		  } else {
-			  logger.debug("No available bridge handler found for bridge: {} .", bridge.getUID());
+			  logger.debug("No available bridge handler found jet. Bridge: {} .", bridge.getUID());
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.vitotronic/src/main/java/org/openhab/binding/vitotronic/handler/VitotronicThingHandler.java]: typo: "yet"

**Comment by kaikreuzer** [comment]
Thanks for this PR. Please also apply the code formatter (this is done automatically, if you switch to the new IDE setup), you still have tabs instead of spaces in the code.

File: addons/binding/org.openhab.binding.vitotronic/ESH-INF/thing/thing-types.xml (hunk 1 above)
Comment (steand) [inline addons/binding/org.openhab.binding.vitotronic/ESH-INF/thing/thing-types.xml]: ups. i don't know why. Maybe a old version. I going to change to "side" again.

**Comment by kaikreuzer** [comment]
Thanks!
